packages/jsii-pacmak/lib/targets/golang/types/class.ts

```diff
@@ -1,63 +1,178 @@
 import { Method, Property, ClassType } from 'jsii-reflect';
 import { CodeMaker } from 'codemaker';
-import { TypeMapper } from './interface';
-import { GoType } from './go-type';
+import { GoTypeRef } from './go-type-reference';
+import { GoType, GoEmitter } from './go-type';
+import { Module } from '../module';
 
-const CLASS_INTERFACE_SUFFIX = 'Iface';
+export class ClassField {
```

Comment:
See https://github.com/aws/jsii/pull/1892#discussion_r470902623

Reply:
This is mostly done for ease of tracking and extracting types and modules that the fields depend on. Makes tracking where to resolve dependencies the same place where type references are instantiated. Same goes for the `InterfaceMethod` and `InterfaceProperty` classes.

Removed this base class though and consolidated all the common structure into a `TypeField` interface that is also used for interfaces.